reports/tasks.py

```diff
@@ -0,0 +1,89 @@
+"""
+Worker tasks for reporting
+"""
+
+# Imports #####################################################################
+
+from datetime import datetime, timedelta
+import logging
+
+from django.conf import settings
+from django.core.mail import EmailMessage
+from django.core.management import call_command
+from huey.api import crontab
+from huey.contrib.djhuey import db_periodic_task
+
+from instance.models.openedx_instance import OpenEdXInstance
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger(__name__)
+
+
+# Constants ###################################################################
+
+TRIAL_INSTANCE_REPORT_SENDER = getattr(settings, 'TRIAL_INSTANCE_REPORT_SENDER',
+                                       settings.DEFAULT_FROM_EMAIL)
+
+
+# Tasks #######################################################################
+
+# Run on the 1st of every month
+@db_periodic_task(crontab(day='1', hour='2', minute='0'))
+def send_trial_instance_report(recipients=settings.TRIAL_INSTANCE_REPORT_RECIPIENTS):
+    """
+    Generate and send a trial instance data report
+
+    This task runs on the first of every month at 2AM
+    """
+
+    # Get instances with an active betatestapplication and
+    # whose InstanceReference has an active OpenEdXAppServer
+    instances = OpenEdXInstance.objects \
+        .exclude(betatestapplication__isnull=True) \
+        .filter(ref_set__openedxappserver_set___is_active=True)
+
+    domains = [instance.external_lms_domain or instance.internal_lms_domain for instance in instances]
+
+    # Start with current date
+    today = datetime.utcnow().date()
+
+    # Get last month by subtracting the day of the month from
+    # the actual date to get the last day of last month
+    # Then replace the day with 1, to get the first day of last month
+    end_of_last_month = (today - timedelta(days=today.day))
+    beginning_of_last_month = end_of_last_month.replace(day=1)
+
+    csv_filename = '/tmp/trial_instances_report.csv'
+
+    call_command(
+        'instance_statistics_csv',
+        out=csv_filename,
+        domains=','.join(domains),
+        start_date=beginning_of_last_month.strftime("%Y-%m-%d"),
+        end_date=end_of_last_month.strftime("%Y-%m-%d")
+    )
```

Comment:
@pcockwell, we should handle any exceptions coming out of this call and send an email notifying the `TRIAL_INSTANCE_REPORT_SENDER` about it.